sqlglot/dialects/presto.py

```diff
@@ -149,6 +149,8 @@ def _sequence_sql(self: generator.Generator, expression: exp.GenerateSeries) ->
 
     sql = self.func("SEQUENCE", start, end, step)
     if isinstance(expression.parent, exp.Table):
+        if expression.parent.alias:
+            expression.parent.args["alias"] = f"_u({expression.parent.alias})"
```

Comment:
This is a bit rough!

1. I'm not sure if mutating the parent expression like this is OK
2. We probably need to ensure that the table alias name (`_u`) isn't taken? If so, we could do this? Like in https://github.com/tobymao/sqlglot/pull/1501/files.

```suggestion
        if expression.parent.alias:
            from sqlglot.optimizer.scope import build_scope
            taken_source_names = set(build_scope(expression.parent_select).selected_sources)
            source_name = find_new_name(taken_source_names, "_u")
            expression.parent.args["alias"] = f"{source_name}({expression.parent.alias})"
```

Reply:
> I'm not sure if mutating the parent expression like this is OK

It's not -- generator methods are not supposed to mutate without first copying. Additionally, the way you tried to mutate the `alias` arg is wrong for two more reasons:

1) You need to store a `TableAlias` with the respective `Identifier` under it that wraps around the target name, instead of a raw string (see example). There are some helper methods for similar things, in `expressions.py`.
2) Mutations in an `Expression` must always be done through the `.set(...)` method, otherwise the resulting AST will be incorrect (i.e. there won't be a link between children / parent expressions).

> We probably need to ensure that the table alias name (_u) isn't taken?

That is correct, for the result to be fully correct you'd have to avoid name clashes. At this point maybe I'd rather create a transformation for this instead of doing it in a `_sql` method (i.e. create a new function in `transforms.py` that does any needed changes in the AST).

I'd suggest to get these changes in with your PR and then I'll look if I can refactor some stuff, e.g. finding all used source names more easily etc.

```python
>>> sqlglot.parse_one("select * from foo a")
(SELECT expressions:
  (STAR ), from:
  (FROM expressions:
    (TABLE this:
      (IDENTIFIER this: foo, quoted: False), alias:
      (TABLEALIAS this:
        (IDENTIFIER this: a, quoted: False)))))
```

Reply:
Perfect, thanks so much for the detailed response. Will try to get this in soon.